src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -3725,12 +3730,13 @@ DataFetchTaskList(uint64 jobId, uint32 taskIdIndex, List *fragmentList)
 		RangeTableFragment *fragment = (RangeTableFragment *) lfirst(fragmentCell);
 		if (fragment->fragmentType == CITUS_RTE_RELATION)
 		{
-			uint64 *shardId = fragment->fragmentReference;
-			StringInfo shardFetchQueryString = ShardFetchQueryString(*shardId);
+			ShardInterval *shardInterval = fragment->fragmentReference;
```

Comment:
I've not done it, because it makes the diff a lot bigger (as one should assert before assigning) ;)